src/KDSoapClient/KDSoapClientInterface.cpp

```diff
@@ -142,21 +144,29 @@ QNetworkRequest KDSoapClientInterfacePrivate::prepareRequest(const QString &meth
     return request;
 }
 
-QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const KDSoapHeaders &headers)
+QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, KDSoapMessage &message, const QString &soapAction, const KDSoapHeaders &headers)
```

Comment:
I think removing const of message is a bad idea. You are now changing the original message, because of the reference.

Reply:
> Please consider adding a unit test for this new functionality.

Many thanks for the review! I have added an unit test and fixed your mentioned issues.